build.py

```diff
@@ -115,6 +115,20 @@ def Exit( self ):
   'See the YCM docs for details on how to use a custom Clangd.' )
 
 
+def FindLatestMSVC():
+    import winreg
+    aReg = winreg.ConnectRegistry(None, winreg.HKEY_LOCAL_MACHINE)
+    msvc = None
+    for i in [17, 16, 15]:
+        try:
+            aKey = winreg.OpenKey(aReg, rf'SOFTWARE\Microsoft\VisualStudio\{i}.0')
+            print(f"Found {i}")
+            msvc = i
+            break
+        except FileNotFoundError:
+            pass
+    return msvc
+
```

Comment:
This is what cmake does: # https://gitlab.kitware.com/cmake/cmake/-/blob/master/Source/cmGlobalVisualStudioGenerator.cxx#L201